x-pack/plugins/enterprise_search/public/applications/workplace_search/views/content_sources/source_logic.ts

```diff
@@ -239,7 +238,8 @@ export const SourceLogic = kea<MakeLogicType<SourceValues, SourceActions>>({
 
       try {
         const response = await HttpLogic.values.http.delete(route);
-        setQueuedSuccessMessage(
+        KibanaLogic.values.navigateToUrl(getSourcesPath(SOURCES_PATH, isOrganization));
```

Comment:
The redirect needs to occur before the toast is flashed. This wasn't working before with `setQueuedSuccessMessage` and should have been redirected before there as well. Verified this is working as expected with toasts